src/lib/asn1/asn1_obj.h

```diff
@@ -118,34 +140,35 @@ class BOTAN_PUBLIC_API(2,0) BER_Object final
 
       BER_Object& operator=(BER_Object&& other) = default;
 
-      bool is_set() const { return m_type_tag != ASN1_Tag::NO_OBJECT; }
+      bool is_set() const { return m_type_tag != ASN1_Type::NO_OBJECT; }
 
-      ASN1_Tag tagging() const { return type_tag() | class_tag(); }
+      uint32_t tagging() const { return type_tag() | class_tag(); }
```

Comment:
I don't like how we lose the type here but don't really see a good way of avoiding it.

Reply:
Yeah my me too. I am open to suggestions.